### ISSUE TITLE
Specify ELB placement by Subnet ID

### ISSUE THREAD

**Issue by tsfrt-vw** [kubernetes-sigs/cluster-api-provider-aws#2296]
/kind bug

**What steps did you take and what happened:**
We are trying to replicate the standard deployment pattern with an ELB and NAT Gateway in 3 public subnets and placing all or our EC2 instances into private subnets.  In this deployment our public subnets do not have an internet-gateway but are connected to a corporate network (via Transit Gateway)and uses corporate routable IP space.  We want to limit IP usage in the public subnet and to preserve the security posture of the standard deployment.  This works with a single control plane node, but not a multi-az deployment because we are unable to influence the placement of the ELB and Control plane nodes.

1.  Create VPC with 6 subnets across 3 AZs: 3 on corporate routable (public), 3 non-routable (private). 

2. Set LB Scheme in AWSCluster to internal 

scheme: internal

2.  Include all subnets in AWSCluster definition (none of them get set as public)

Three Public subnets get listed first, followed by 3 private.  This seems to locate the ELB in the public subnets, regardless of tagging. (This could be a point of confusion, can tagging solve this internal-elb or cluster tags do not seem to do it?)

3. For 3 AZ Machine Deployments, specify the Subnet ID in each AWSMachineTemplate that corresponds to a given failure domain (3 of these).

```
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
kind: AWSMachineTemplate
metadata:
  name: CLUSTER_NAME
spec:
  template:
    spec:
      instanceType: TYPE
      iamInstanceProfile: "nodes"
      sshKeyName: "NAME"
      ami:
        id: "AMI"
      rootVolume:
        size: 80
      subnet:
        id: <Set this to the subnet in the AZ that corresponds to MD>

```
4.  KubeadmControlPlane only references a single infrastructure template, so we can not specify Subnet ID for this AWSMachineTemplate or it will result in the subnet id not being found (in 2 of 3 AZs).  

If we don't specify a Subnet ID here, the control plane nodes are placed into the same subnet as my ELB.  While workers are placed in their specific subnets.

In summary:
-ELB is in 3 public subnets
-Control Plane nodes are in public subnets
-Worker Nodes are in private subnets

**What did you expect to happen:**
-ELB is created in my public subnet (routable one in my case)
-All EC2 instances are created in my private subnets (non-routable)

**Anything else you would like to add:**
This approach works with a single control plane node because we can set the subnet id in the AWSMachineTemplate for the control plane and be sure the AZ will match up with the Subnet ID.

**Environment:**

- Cluster-api-provider-aws version: v0.6.3
- Kubernetes version: (use `kubectl version`):  v1.19.3
- OS (e.g. from `/etc/os-release`): amazon_linux:2
